### PR TITLE
[fix]: [PL-29315]: Added error handling in ssh secret validation flow.

### PR DESCRIPTION
**Pull request by nishantsaini20**
## Describe your changes
The `executeSyncTask` method throws `DelegateServiceDriverException` when delegates are not available. This exception was not handled properly to propagate correct error info to the user. The method call is now wrapped with try catch block and correct exception is thrown which propagates useful error message to the user.
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40371)

### CODE REVIEW

**Review by shashank-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40371)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGSecretServiceV2Impl.java [hunk 1]
```diff
@@ -302,16 +307,27 @@ private SecretValidationResultDTO validateSecretSSHKey(String accountIdentifier,
...
-    } else if (delegateResponseData instanceof SSHConfigValidationTaskResponse) {
-      return buildBaseConfigValidationTaskResponse((SSHConfigValidationTaskResponse) delegateResponseData);
+    try {
+      DelegateResponseData delegateResponseData = this.delegateGrpcClientWrapper.executeSyncTask(delegateTaskRequest);
+      if (delegateResponseData instanceof RemoteMethodReturnValueData) {
+        return buildRemoteMethodResponse((RemoteMethodReturnValueData) delegateResponseData);
+      } else if (delegateResponseData instanceof SSHConfigValidationTaskResponse) {
+        return buildBaseConfigValidationTaskResponse((SSHConfigValidationTaskResponse) delegateResponseData);
+      }
+    } catch (DelegateServiceDriverException ex) {
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/NGSecretServiceV2Impl.java:317]: Is this exception for No Delegates in Account or No eligible delegates to perform this task?

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40371)* status: 2 file(s) to review.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40371)* status: :shipit: complete! ✓

**Pull request merged**